### DISCUSSION TITLE
Keep cell on fixed position relative to viewport

### DISCUSSION THREAD

**Discussion by CuMarioStarlinger** [clientIO/joint#2173] [Q&A]
### Introduction

How can you set the position of a cell fixed relative to the viewport?
I want to set the position of a cell fixed even changing the viewport (dragging around the paper). 
I tried to achieve this by listening on the 'blank:pointermove' event and changing the position of the cell with following code: 

```javascript
cell.set('position', {
    x: cellPos.x - ((<MouseEvent>evt.originalEvent).movementX / paper.scale().sx),
    y: cellPos.y - ((<MouseEvent>evt.originalEvent).movementY / paper.scale().sy),
  }
``` 

But this does not work very smooth. Sometimes the cell jumps to a wrong position. 

Is there a better way to do this?

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS, JointJS+

**Comment by kumilingus** [comment]
I guess this is similar to what you have:
```js
const { paperScroller, paper, graph } = this;
const el = graph.getElements()[0];
paperScroller.on('scroll', () => positionEl(el));

function positionEl(el) {
    const { left, top } = paperScroller.el.getBoundingClientRect();
    const localPoint = paper.clientToLocalPoint({ x: left, y: top });
    el.position(localPoint.x, localPoint.y, { async: false });
}
```
